File: LeetCode/Easy/1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.py
class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:
        def countBits(n):
            count = 0
            while n:
                count += n & 1
                n >>= 1
            return count

        return sorted(arr, key=lambda x: (countBits(x), x))

File: LeetCode/Easy/1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        sort(arr.begin(), arr.end(), customComparator);
        return arr;
    }

    static int countBits(int n) {
        int count = 0;
        while (n) {
            count += n & 1;
            n >>= 1;
        }
        return count;
    }

    static bool customComparator(int &a, int &b) {
        int countA = countBits(a);
        int countB = countBits(b);

        if (countA == countB)
            return a < b;

        return countA < countB;
    }
};